disttask/framework/storage/task_table.go

```diff
@@ -339,3 +355,35 @@ func (stm *TaskManager) GetSchedulerIDsByTaskID(taskID int64) ([]string, error)
 
 	return instanceIDs, nil
 }
+
+// UpdateGlobalTaskAndAddSubTasks update the global task and add new subtasks
+func (stm *TaskManager) UpdateGlobalTaskAndAddSubTasks(gTask *proto.Task, subtasks []*proto.Subtask, isSubtaskRevert bool) error {
+	return stm.withNewTxn(func(se sessionctx.Context) error {
+		_, err := execSQL(stm.ctx, se, "update mysql.tidb_global_task set state = %?, dispatcher_id = %?, step = %?, state_update_time = %?, concurrency = %? where id = %?",
+			gTask.State, gTask.DispatcherID, gTask.Step, gTask.StateUpdateTime.UTC().String(), gTask.Concurrency, gTask.ID)
+		if err != nil {
+			return err
+		}
+
+		failpoint.Inject("MockUpdateTaskErr", func(val failpoint.Value) {
+			if val.(bool) {
+				failpoint.Return(errors.New("updateTaskErr"))
+			}
+		})
+
+		subtaskState := proto.TaskStatePending
+		if isSubtaskRevert {
+			subtaskState = proto.TaskStateRevertPending
+		}
+
+		for _, subtask := range subtasks {
+			_, err = execSQL(stm.ctx, se, "insert into mysql.tidb_background_subtask(task_key, exec_id, meta, state, type, checkpoint) values (%?, %?, %?, %?, %?, %?)",
```

Comment:
Could we consider batch insert or add a  TODO?

Reply:
added some comments